### DISCUSSION TITLE
terrain layer loadOptions

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#5995] [Q&A]
Hello,

I'm trying to pass custom headers only to the mesh layer, not the tiles layer. Is it possible? Or should this approach work?

```
_subLayerProps: {
  tiles: {

  },
  mesh: {
    loadOptions: {
      fetch: {
        headers: { ... }
      }
    }
  }
}
```

headers are missing in both requests with ^^ setting.

When I set loadOptions in TerrainLayer, headers are used in both layers. I need to have a better control over both requests. I think it's not issue in loaders.gl, because TerrainWorkerLoader loads just mesh layer.

using 8.5.0-beta.1

Thanks

**Comment by Pessimistress** [comment]
> When I set loadOptions in TerrainLayer, headers are used in both layers.

What do you mean by that? The mesh sublayer does not make any request on its own. All requests are issued by the TerrainLayer.

**Reply by l3hgs-brent** [reply]
We've also encountered this issue.

We are attempting to use the [TerrainLayer class](https://deck.gl/docs/api-reference/geo-layers/terrain-layer) with both the required 'elevationData' and optional 'texture' data options.  There are two different data sources for this data.

One of the data sources requires an HTTP Authorization request header.  We can provide the authorization credentials for this data source via a TerrainLayer 'loadOptions' property (per the ["fetch data with credentials" example](https://deck.gl/docs/developer-guide/loading-data#example-fetch-data-with-credentials)).  Unfortunately, these same 'loadOptions' are used in requests to the second data source, where they are not supported and rejected with 400 Bad Request responses.

Our use case requires being able to specify differing HTTP request headers for 'elevationData' fetch requests and 'texture' data fetch requests.  Is there any way to accommodate this? @radoslavirha suggests above that use of _subLayerProps for this does not work.
